### PR TITLE
Add built-in Paper datapack

### PR DESCRIPTION
**Pull request by jpenilla**
- Adds an EntityTypeTag for each MobCategory

Not sure whether these tags should be added to API or not.
The purpose is to allow filtering entity selectors by these tags.

### CODE REVIEW

File: patches/server/0752-Add-Paper-built-in-datapack.patch [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
++        Files.createDirectories(datapackFolder);
++
++        Files.writeString(
++            packMeta(datapackFolder),
++            "{\n"
++                + "    \"pack\": {\n"
++                + "        \"description\": \"Paper built-in datapack\",\n"
++                + "        \"pack_format\": " + SharedConstants.getCurrentVersion().getPackVersion(PackType.DATA) + "\n"
++                + "    }\n"
++                + "}\n"
```
Comment (imDaniX) [inline patches/server/0752-Add-Paper-built-in-datapack.patch]: Use the text block from Java 15?

File: patches/server/0752-Add-Paper-built-in-datapack.patch [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
++
++@DefaultQualifier(NonNull.class)
++public final class PaperEntityTypeTagsProvider extends EntityTypeTagsProvider {
++    private static final StaticTagHelper<EntityType<?>> HELPER = new StaticTagHelper<>(Registry.ENTITY_TYPE_REGISTRY, "tags/entity_types"); // args copied from superclass
++
++    public static final Map<MobCategory, Tag.Named<EntityType<?>>> CATEGORY_TAGS = new HashMap<>() {{
++        for (final MobCategory category : MobCategory.values()) {
++            this.put(category, HELPER.bind("paper:category/" + category.getName()));
++        }
++    }};
```
Comment (Proximyst) [inline patches/server/0752-Add-Paper-built-in-datapack.patch]: Stream it instead

**Review by Machine-Maker — APPROVED** [review]

File: patches/server/0752-Add-Paper-built-in-datapack.patch [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
++    }
++
++    public static void installOrUpdateDatapack(final Path datapackFolder) throws IOException {
++        Files.createDirectories(datapackFolder);
++
++        Files.writeString(
++            packMeta(datapackFolder),
++            "{\n" +
++                "    \"pack\": {\n" +
++                "        \"description\": \"Paper built-in datapack\",\n" +
```
Comment (MiniDigger) [inline patches/server/0752-Add-Paper-built-in-datapack.patch:42]:
```suggestion
+                "        \"description\": \"Paper's built-in datapack\",\n" +
```
?

**Comment by Machine-Maker** [comment]
I’m working on registering custom tags (sent to player, in suggestions, etc) so idk if those is will be needed then, unless it’s useful to have for something else.

**Pull request closed without merging**